test/common/stats/stats_impl_test.cc

```diff
@@ -69,5 +83,249 @@ TEST(StatsMacros, All) {
+      "http.dynamodb.table.capacity",
+      {dynamo_http_prefix, dynamo_table, dynamo_operation, dynamo_partition});
+
+  // GRPC Http1.1 Bridge
+  Tag grpc_cluster;
+  grpc_cluster.name_ = tag_names.CLUSTER_NAME;
+  grpc_cluster.value_ = "grpc_cluster";
+
+  Tag grpc_service;
+  grpc_service.name_ = tag_names.GRPC_BRIDGE_SERVICE;
+  grpc_service.value_ = "grpc_service_1";
+
+  Tag grpc_method;
+  grpc_method.name_ = tag_names.GRPC_BRIDGE_METHOD;
+  grpc_method.value_ = "grpc_method_1";
+
+  regex_tester.testRegex("cluster.grpc_cluster.grpc.grpc_service_1.grpc_method_1.success",
+                         "cluster.grpc.success", {grpc_cluster, grpc_method, grpc_service});
+
+  // Virtual host and cluster
+  Tag vhost;
+  vhost.name_ = tag_names.VIRTUAL_HOST;
+  vhost.value_ = "vhost_1";
+
+  Tag vcluster;
+  vcluster.name_ = tag_names.VIRTUAL_CLUSTER;
+  vcluster.value_ = "vcluster_1";
+
+  Tag response_code_class;
+  response_code_class.name_ = tag_names.RESPONSE_CODE_CLASS;
+  response_code_class.value_ = "2xx";
+
+  Tag response_code;
+  response_code.name_ = tag_names.RESPONSE_CODE;
+  response_code.value_ = "200";
+
+  regex_tester.testRegex("vhost.vhost_1.vcluster.vcluster_1.upstream_rq_2xx",
+                         "vhost.vcluster.upstream_rq", {vhost, vcluster, response_code_class});
+  regex_tester.testRegex("vhost.vhost_1.vcluster.vcluster_1.upstream_rq_200",
+                         "vhost.vcluster.upstream_rq", {vhost, vcluster, response_code});
+
+  // User agent
+  Tag user_agent;
+  user_agent.name_ = tag_names.HTTP_USER_AGENT;
+  user_agent.value_ = "ios";
+
+  regex_tester.testRegex("http.egress_dynamodb_iad.user_agent.ios.downstream_cx_total",
+                         "http.user_agent.downstream_cx_total", {user_agent, dynamo_http_prefix});
+
+  // Client SSL Prefix
+  Tag client_ssl;
+  client_ssl.name_ = tag_names.CLIENTSSL_PREFIX;
+  client_ssl.value_ = "clientssl_prefix";
+
+  regex_tester.testRegex("auth.clientssl.clientssl_prefix.auth_ip_white_list",
+                         "auth.clientssl.auth_ip_white_list", {client_ssl});
+
+  // TCP Prefix
+  Tag tcp_prefix;
+  tcp_prefix.name_ = tag_names.TCP_PREFIX;
+  tcp_prefix.value_ = "tcp_prefix";
+
+  regex_tester.testRegex("tcp.tcp_prefix.downstream_flow_control_resumed_reading_total",
+                         "tcp.downstream_flow_control_resumed_reading_total", {tcp_prefix});
+
+  // Fault Downstream Cluster
+  Tag fault_connection_manager;
+  fault_connection_manager.name_ = tag_names.HTTP_CONN_MANAGER_PREFIX;
+  fault_connection_manager.value_ = "fault_connection_manager";
+
+  Tag fault_downstream_cluster;
+  fault_downstream_cluster.name_ = tag_names.FAULT_DOWNSTREAM_CLUSTER;
+  fault_downstream_cluster.value_ = "fault_cluster";
+
+  regex_tester.testRegex("http.fault_connection_manager.fault.fault_cluster.aborts_injected",
+                         "http.fault.aborts_injected",
+                         {fault_connection_manager, fault_downstream_cluster});
+}
```

Comment:
An alternative testing strategy might be to run the tag extraction after an integration test, dump the outputs, manually verify create a "golden" state and then compare. I don't think you need to do this, there are already a bunch of reasonable tests, but this kind of testing would sanity check what happens when we run on real-ish data.

Reply:
Good point.  I tried this out and found a problem with my expectations (based on the docs) - "listener." is followed by `<ip>_<port>` rather than just port, so it wasn't being captured.  Not sure the best way to do this though... read in a text file?  Do we have other tests that work this way?  I want to make it as low-maintenance as possible, meaning I would rather people not have to re-generate the output every time they add a stat.  That sort of setup would make it easy for people to accidentally ignore an issue they created because it becomes the normal process to modify this test each time anyone touches stats.

Reply:
We don't really have tests that do this today. I think maybe just adding to one of the integration tests a validation of the full extraction would be useful and give the same benefit as well.